ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/discovery/TimelineMetricMetadataManager.java

```diff
@@ -667,13 +681,17 @@ public TimelineMetric getMetricFromUuid(byte[] uuid) {
           }
           TimelineMetric metric = new TimelineMetric();
           metric.setMetricName(metricName);
-          metric.setAppId(appId);
-          metric.setInstanceId(instanceId);
           for (String hostname : sanitizedHostNames) {
             metric.setHostName(hostname);
-            byte[] uuid = getUuid(metric, false);
-            if (uuid != null) {
-              uuids.add(uuid);
+            for (String a : sanitizedAppIds) {
+              for (String i : sanitizedInstanceIds) {
+                metric.setAppId(a);
+                metric.setInstanceId(i);
+                byte[] uuid = getUuid(metric, false);
+                if (uuid != null) {
```

Comment:
Shouldn't this if have an else which logs an error?

Reply:
The else is not an ERROR. It just means we have not receieved a metric with this combination of IDs (name + appId + instanceId). Also, the misses are logged in the internal getUuid call.